File: 98-validate-binary-search-tree/98. Validate Binary Search Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(TreeNode* root, vector<int>& sample) {
        if (root == NULL) {
            return; 
        }
        helper(root -> left, sample);
        sample.push_back(root -> val);
        helper(root -> right, sample); 
    }
    bool isValidBST(TreeNode* root) {
        vector<int> sample; 
        helper(root, sample); 
        for (int i = 0; i < sample.size() - 1; i++) {
            if (sample[i] >= sample[i + 1]) {
                return false; 
            }
        }
        return true; 
    }
};

File: 98-validate-binary-search-tree/98. Validate Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder_collector(self, root, collector):
        if(root == None):
            return 
        self.inorder_collector(root.left, collector)
        collector.append(root.val)
        self.inorder_collector(root.right, collector)
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        collector = []
        self.inorder_collector(root, collector)
        for i in range(0, len(collector) - 1):
            if(collector[i] >= collector[i + 1]):
                return False
        return True
        


        